cmake/features/DOCS.cmake

```diff
@@ -1,9 +1,9 @@
 ################################################################################
 # documentation
-if( ENABLE_DOCS ) 
-  find_package(Latex)
+if( ENABLE_DOCS )
+  find_package(LATEX REQUIRED COMPONENTS PDFLATEX BIBTEX OPTIONAL_COMPONENTS MAKEINDEX HTLATEX)
```

Comment:
Why optional for MAKEINDEX, HTLATEX ?

Regardless, this also has to then be reflected in dox/user-guide/CMakeLists.txt so that `atlas-user-guide-html` is depending on HTLATEX.

Reply:
@wdeconinck Some HPC's do not have a complete Latex installation.  They contain the basic Latex compilers e.g. pdflatex, bibtex, but not htlatex.  makeindex is used in the Doxyfile.  At a minimum, the user can build the PDF user-guide.
In doing so, I  wrapped the creation of html documentation  in a `if (LATEX_HTLATEX_FOUND)` block. 
I am happy to remove the `OPTIONAL_COMPONENTS` flag.

I have also simplified the logic in the Doxygen documentation, and relying on `FindDoxygen.cmake` to locate the `DOXYGEN_EXECUTABLE`.  Please review.